test/unit/collection-view.spec.js

```diff
@@ -1318,4 +1318,46 @@ describe('collection view', function() {
       expect(this.childView.$el).to.contain.$text('bar');
     });
   });
+
+  describe('when a collection view has been rendered but not shown', function() {
+    beforeEach(function() {
+      this.ChildView = Backbone.Marionette.ItemView.extend({
+        onBeforeShow: function() {},
+        onShow: function() {},
+        template: _.template('<%= foo %>')
+      });
+
+      this.CollectionView = Backbone.Marionette.CollectionView.extend({
+        childView: this.ChildView
+      });
+
+      this.sinon.spy(this.ChildView.prototype, 'onBeforeShow');
+      this.sinon.spy(this.ChildView.prototype, 'onShow');
+
+      this.model1 = new Backbone.Model({foo: 1});
+      this.model2 = new Backbone.Model({foo: 2});
+      this.collection = new Backbone.Collection([this.model1, this.model2]);
+      this.collectionView = new this.CollectionView({
+        collection: this.collection
+      });
+      $('body').append(this.collectionView.el);
```

Comment:
in another commit that is cool, the call count, context, and argument check is something we started doing a while ago, it makes our tests much more explicit and hardened against regressions.